app/models/importers/hmis_twenty_twenty/base.rb

```diff
@@ -475,6 +483,44 @@ def clean_source_file destination_path:, read_from:, klass:
       write_to.close
     end
 
+    # We sometimes see very odd dates, this will attempt to make them sane.
+    # Since most dates should be not too far in the future, we'll check for anything less
+    # Than a year out
+    private def fix_date_format(string)
+      return unless string
+
+      # Sometimes dates come in mm-dd-yyyy and Ruby Date really doesn't like that.
+      if /\d\d-\d\d-\d\d\d\d/.match?(string)
+        month, day, year = string.split('-')
+        return "#{year}-#{month}-#{day}"
+      end
+      d = Date.parse(string)
```

Comment:
`d = Date.parse(string, false)` to disable the Y2K parsing and then handle the cutoff yourself?